next/components/ui/VisibilityIntersection/VisibilityIntersectionWrapper.tsx

```diff
@@ -0,0 +1,20 @@
+import { FC, ReactNode } from 'react'
+
+import { useIntersectionObserver } from './useIntersectionObserver'
+
+interface Props {
+  children: ReactNode
+
+  onHandleVisibility(): void
+}
+
+export const VisibilityIntersectionWrapper: FC<Props> = ({ children, onHandleVisibility }) => {
```

Comment:
We usually write components like this:
```
const VisibilityIntersectionWrapper = ({ children, onHandleVisibility }: Props) => {
...
}

export default VisibilityIntersectionWrapper
```

But if there is a reason to use `FC<Props>` style than okay :)